front/components/app/Deploy.tsx

```diff
@@ -37,6 +35,162 @@ const cleanUpConfig = (config: RunConfig) => {
   return JSON.stringify(c);
 };
 
+const DEFAULT_INPUTS = [{ hello: "world" }];
+
+interface ViewAppAPIModalProps {
+  owner: WorkspaceType;
+  app: AppType;
+  run: RunType;
+  inputs?: unknown[];
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export function ViewAppAPIModal({
+  owner,
+  app,
+  run,
+  inputs = DEFAULT_INPUTS,
+  isOpen,
+  onClose,
+}: ViewAppAPIModalProps) {
+  const cURLRequest = (type: "run") => {
```

Comment:
`nit`: we could probably share this as it's also being used in `ViewFolderAPIModal`

Reply:
Not really, different request right?

Reply:
Yeah but basically making a curl request. We could have a utility function for this?

Reply:
There's not much to share IMHO :)